java-strings/src/test/java/com/baeldung/string/SubstringPalindromeUnitTest.java

```diff
@@ -0,0 +1,83 @@
+package com.baeldung.string;
+
+import static org.junit.Assert.assertEquals;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Test;
+
+public class SubstringPalindromeUnitTest {
+
+    private static final String INPUT_BUBBLE = "bubble";
+    private static final String INPUT_CIVIC = "civic";
+    private static final String INPUT_INDEED = "indeed";
+    private static final String INPUT_ABABAC = "ababac";
+
+    Set<String> EXPECTED_PALINDROME_BUBBLE = new HashSet<String>() {
+        {
+            add("b");
+            add("u");
+            add("l");
+            add("e");
+            add("bb");
+            add("bub");
+        }
+    };
+
+    Set<String> EXPECTED_PALINDROME_CIVIC = new HashSet<String>() {
+        {
+            add("civic");
+            add("ivi");
+            add("i");
+            add("c");
+            add("v");
+        }
+    };
+
+    Set<String> EXPECTED_PALINDROME_INDEED = new HashSet<String>() {
+        {
+            add("i");
+            add("n");
+            add("d");
+            add("e");
+            add("ee");
+            add("deed");
+        }
+    };
+
+    Set<String> EXPECTED_PALINDROME_ABABAC = new HashSet<String>() {
+        {
+            add("a");
+            add("b");
+            add("c");
+            add("aba");
+            add("bab");
+            add("ababa");
+        }
+    };
+
+    private SubstringPalindrome palindrome = new SubstringPalindrome();
+
+    @Test
+    public void whenApproachUsingManachersAlgo_shouldBePalindrome() {
```

Comment:
These are pretty good names for tests, but let's strictly follow the BDD-style naming convention. For this one, it could be _whenUsingManachersAlgorithm_thenFindsAllPalindromes_. Use a similar name for the other tests.

Reply:
Ok sure